app/controllers/patients_controller.rb

```diff
@@ -1,17 +1,25 @@
 class PatientsController < ApplicationController
   before_action :set_patient, only: [:show, :edit, :update, :destroy]
-  before_action :check_access, only: [:show, :edit, :update, :destroy] 
+  before_action :check_access, only: [:show, :edit, :update, :destroy]
 
   # GET /patients
   # GET /patients.json
   def index
     user = current_user
+    group = user.groups
     sub_ids = []
+    @patients = []
     if not current_user.admin
       if params[:result] == "true"
         @patients = user.patients.where(result: true).order(:case_id).page(params[:page]).per(25)
+        group.each do |g|   
+          @patients = g.patients.where(result: true).order(:case_id).page(params[:page]).per(25)
```

Comment:
You only assign the patients in the last group.

Reply:
I tested it and somehow the @patients object seems to be appended and not overwritten. I'll check again when I get back to this feature with more scenarios

Reply:
![screenshot from 2018-08-06 16-34-39](https://user-images.githubusercontent.com/8877903/43722911-b328fab4-9996-11e8-851b-e745c830344d.png)

Reply:
Then print it out here, show me how it can be appended? If you didn't do it, just say it.

Reply:
Here you go. [Hope u can see, this is the best i can get with team viewer]

 This is what i am talking about. For group 1, i have 2 users. The user jean.tori has only a user patient(which was uploaded before creating group) and the user has 3 patients that are added to group_patients after group creation. In the list i can see all 4. 
the @patient object is being appended here, not sure why, which is what I said. I will further test it with the feature.

I don't have to lie about testing my own code.

![patientlist_groupanduserpatients](https://user-images.githubusercontent.com/22116058/43735539-1f88b61c-99bb-11e8-9356-66740c21e196.png)

Reply:
This for loop is to go through every group this user belongs to. g.patients
You assign the patients from each member by this.
```
def assign_group_patients
    # This automatically creates the UserGroup record
    user = @group.users
    user.each do |u|
      email = u.email
      submitter = Submitter.find_by_email(email)
      if submitter != nil
        submitter.patient.each do |patient|
          if not @group.patients.exists?(patient.id)
            @group.patients << patient
          end
        end
      end
    end
  end
```
And list all the patients from every group which this user belongs to.
If user A belongs to Group A and Group B. Only the patients of group B will be in the variable @patients because you assign to the same variable in the for loop instead of append them.
For loop here is not for go through patients in the same group, but go through every group the user belongs to.
```
group = user.groups
group.each do |g|   
          @patients = g.patients.where(result: true).order(:case_id).page(params[:page]).per(25)
end
```
Every code you put in this pull request should be tested. This is also part of UI. If you didn't test for this scenario, please remove this feature from this pull request.